src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -169,19 +194,19 @@ function ConfirmRejectionModal ({ transaction, handleClose, declineTransaction }
   </Modal>
 }
 
-export function RenderNickname ({ agentId }) {
-  const { loading, error, data } = useQuery(HolofuelCounterpartyQuery, {
+export function RenderNickname ({ agentId, copyId }) {
+  const { loading, error, data: { holofuelCounterparty = {} } = {} } = useQuery(HolofuelCounterpartyQuery, {
     variables: { agentId }
   })
 
-  if (loading) return <>Loading...</>
-
-  if (error) {
+  if ((loading || error) && !copyId) return <>{presentAgentId(agentId)}</>
+  else if ((loading || error) && copyId) {
     return <CopyAgentId agent={{ id: agentId, nickname: '' }}>
       {presentAgentId(agentId)}
     </CopyAgentId>
-  }
-  return <CopyAgentId agent={data.holofuelCounterparty}>
-    {data.holofuelCounterparty.nickname}
-  </CopyAgentId>
+  } else if (holofuelCounterparty.nickname && copyId) {
+    return <CopyAgentId agent={holofuelCounterparty}>
+      {holofuelCounterparty.nickname}
+    </CopyAgentId>
+  } else return <>{holofuelCounterparty.nickname}</>
```

Comment:
still think i prefer the approach of just wrapping this component in a <CopyAgentId> when we need to, rather than trying to build the functionality in here. This kind of compound if statement is harder to read and think about, and error prone.

Reply:
The reason I wrap the component here, is because we don't have access to the holofuelConterparty nickname outside of the RenderNickname Component, which CopyAgentId needs as a value for it's `agent` prop.

Open to other ideas though.